clang-tools-extra/docs/clang-tidy/checks/modernize/use-designated-initializers.rst

```diff
@@ -0,0 +1,54 @@
+.. title:: clang-tidy - modernize-use-designated-initializers
+
+modernize-use-designated-initializers
+=====================================
+
+Finds initializer lists for aggregate type that could be written as
+designated initializers instead.
+
+With plain initializer lists, it is very easy to introduce bugs when adding
+new fields in the middle of a struct or class type. The same confusion might
+arise when changing the order of fields.
+
+C++20 supports the designated initializer syntax for aggregate types.
+By applying it, we can always be sure that aggregates are constructed correctly,
+because every variable being initialized is referenced by name.
+
```

Comment:
Add some comment about -Wc99-designator (this is about mixed), and -Wc++20-designator (this is about using designated initializer in older C++ standard)